### PR TITLE
feat(ui): add dark mode

### PR DESCRIPTION
**Pull request by RichardKruemmel**

### CODE REVIEW

File: web/src/pages/_app.tsx [hunk 1]
```diff
@@ -87,10 +88,17 @@ const MyApp: AppType<{ session: Session | null }> = ({
...
           <SessionProvider session={session} refetchOnWindowFocus={true}>
             <DetailPageListsProvider>
-              <Layout>
-                <Component {...pageProps} />
-                <UserTracking />
-              </Layout>
+              <ThemeProvider
+                attribute="class"
+                defaultTheme="light"
+                enableSystem
```
Comment (marcklingen) [inline web/src/pages/_app.tsx:94]:
```suggestion
                enableSystem={false}
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -290,6 +290,14 @@ When a new release is tagged on the `main` branch (excluding prereleases), it tr
 1. The Docker image is published to GitHub Packages with the version number and `latest` tag.
 2. The deployment is carried out on Langfuse Cloud. This is done by force pushing the `main` branch to the `production` branch during every release, using the [`release.yml`](.github/workflows/release.yml) GitHub Action.
 
+## Theming
+
+At Langfuse, we utilize CSS variables to manage our theme settings across the platform.
+
+Our theming approach is utilizing separate CSS variables for background (--background) and foreground (--foreground) colors. We recommend using HSL values for these colors to enhance user interface consistency and customization capabilities.
```
Comment (marcklingen) [inline CONTRIBUTING.md:297]: I'd directly reference the color palette used by shadcn/ui as we probably fully rely on that naming/convention. What do you think?

**Review by marcklingen — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (RichardKruemmel) [reply CONTRIBUTING.md]: Makes sense. I added a change explicitly mentioning and linking shadcn theming docs.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -290,6 +290,34 @@ When a new release is tagged on the `main` branch (excluding prereleases), it tr
...
+- Border Color: `border`
+- Primary Text: `primary`
+- Secondary Text: `muted-foreground`
+- Hover States: Use `primary-foreground` or `accent`
+- Background: `background`
+- Main Content: `foreground`
+- Primary Color: `primary`
+- Accents: `primary-accent`
+- Selectable Items: `input`
+- Link Background: `muted-indigo`
```
Comment (marcklingen) [inline CONTRIBUTING.md:315]: can we switch to some template here as well? I though we use some [template]\80 for these

also, other colors (red, yellow, green) missing

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -290,6 +290,34 @@ When a new release is tagged on the `main` branch (excluding prereleases), it tr
...
+At Langfuse, we utilize CSS variables to manage our theme settings across the platform.
+
+Our approach leverages separate CSS variables for backgrounds (--background) and foregrounds (--foreground), fully adhering to the [shadcn/ui](https://ui.shadcn.com/docs/theming) color conventions. We recommend using HSL values for these colors to enhance consistency and customization.
+
+### Adding New Colors
+
+1. Global Definitions: Add new CSS variable definitions in the global.css file.
+2. Tailwind Configuration: Reflect these new colors in the tailwind.config.js to maintain alignment with Tailwind's utility classes.
+
+### Utility Classes
```
Comment (marcklingen) [inline CONTRIBUTING.md:304]: I'd add a couple of more words to each with examples, to me it is not 100% clear what accent, main content etc means. All contributors need to easily understand which colors to apply to which parts of their components

File: web/src/components/layouts/layout.tsx [hunk 5]
```diff
@@ -376,11 +381,12 @@ export default function Layout(props: PropsWithChildren) {
...
                           onClick={() => void item.onClick()}
                           className={cn(
-                            active ? "bg-gray-50" : "",
-                            "block cursor-pointer px-3 py-1 text-sm text-gray-900",
+                            active ? "bg-primary-foreground" : "",
+                            "flex cursor-pointer items-center justify-between px-2 py-0.5 text-sm leading-6 text-primary",
                           )}
                         >
                           {item.name}
+                          {item.name === "Theme" && <ThemeToggle />}
```
Comment (marcklingen) [inline web/src/components/layouts/layout.tsx:389]: just add as custom element outside of userNavigation.map, this way this ifelsehas a condition on the item being in the list, not clear what is happening here to someone reading this

File: web/src/components/layouts/layout.tsx [hunk 6]
```diff
@@ -440,11 +446,12 @@ export default function Layout(props: PropsWithChildren) {
...
                         onClick={() => void item.onClick()}
                         className={cn(
-                          active ? "bg-gray-50" : "",
-                          "block cursor-pointer px-3 py-1 text-sm text-gray-900",
+                          active ? "bg-primary-foreground" : "",
+                          "flex cursor-pointer items-center justify-between px-2 py-1 text-sm text-primary",
                         )}
                       >
                         {item.name}
+                        {item.name === "Theme" && <ThemeToggle />}
```
Comment (marcklingen) [inline web/src/components/layouts/layout.tsx:454]: see above

File: web/src/components/ui/CodeJsonViewer.tsx [hunk 7]
```diff
@@ -35,6 +44,7 @@ export function JSONView(props: {
           <React18JsonView
             src={parsedJson}
             theme="github"
+            dark={theme === "dark"}
```
Comment (marcklingen) [inline web/src/components/ui/CodeJsonViewer.tsx:47]: uhh, nice!

**Review by marcklingen — APPROVED** [review]

**Pull request merged**